x-pack/metricbeat/module/mssql/performance/performance.go

```diff
@@ -113,11 +117,14 @@ WHERE  counter_name = 'SQL Compilations/sec'
 		row.objectName = strings.TrimSpace(row.objectName)
 
 		if row.counterName == "Buffer cache hit ratio" {
-			mapStr[row.counterName] = fmt.Sprintf("%v", float64(*row.counterValue)/100)
+			buffer_cache_hit_ratio = *row.counterValue
+		} else if row.counterName == "Buffer cache hit ratio base" {
+			buffer_cache_hit_ratio_base = *row.counterValue
 		} else {
 			mapStr[row.counterName] = fmt.Sprintf("%v", *row.counterValue)
 		}
 	}
+	mapStr[BufferCacheHitRatio] = fmt.Sprintf("%v", float64(buffer_cache_hit_ratio/buffer_cache_hit_ratio_base))
```

Comment:
Are you sure `float64(buffer_cache_hit_ratio/buffer_cache_hit_ratio_base)` is the correct logic?

If I am reading this code correctly, you've declared buffer_cache_hit_ratio and buffer_cache_hit_ratio_base and they have zero values by default i.e., 0 for int64 type.

With setting values in `if`, `else-if`, `else` block seems only one of the two vars will be updated and not both.

That's why I think there's a bug in the logic. If buffer_cache_hit_ratio_base is 0 you divide, program fails.

Reply:
null check value added.